.github/workflows/cppcheck.yml

```diff
@@ -0,0 +1,32 @@
+name: cppcheck Static Analysis
+
+# Run cppcheck on src_c changes to main branch, or any PR to main.
+on:
+  push:
+    branches: main
+    paths:
+      - 'src_c/**'
+
+  pull_request:
+    branches: main
+    paths:
+      - 'src_c/**'
+
+# TODO: Any more static checkers can be added here
+jobs:
+  run-cppcheck:
+    runs-on: ubuntu-20.04
```

Comment:
cppcheck! Very nice.

I wonder though if we should use the latest release? https://github.com/danmar/cppcheck#cmake

The one on ubuntu 20.04 is a bit old.

Reply:
I guess there's no easy to install package available... so building it from source would take too long. I looked for a pypi package too, and couldn't find one.

Probably sticking to the ubuntu one is fine IMHO.

Reply:
Yeah, it should be fine for now. We can always update this runner to use Ubuntu 22.04 later when it starts rolling to GH actions runners in a few months

Reply:
Wait actually... CPPCheck provides windows installers on their [github releases](https://github.com/danmar/cppcheck/releases).
Ubuntu 20.04 one is still at 1.9.0 while the latest version is 2.6

Perhaps for some extra safety, that could be an additional check too